webapp/src/dialogs.tsx

```diff
@@ -714,3 +717,74 @@ export function promptTranslateBlock(blockid: string, blockTranslationIds: strin
+        <div className="column sixteen wide instructions">
+            <div className="ui grid">
+                <div className="row">
+                    <div className="column">
+                        <div className="ui two column grid padded">
+                            <div className="column">
+                                <div className="ui">
+                                    <div className="content">
+                                        <div className="description">
+                                            {lf("Your code is being downloaded as a {1} file. You can drag this file to your {0} using your computer's file explorer.", boardName, fileExtension)}
+                                        </div>
+                                        <div className="download-callout">
+                                            <label className="ui purple ribbon large label">{lf("New!")}</label>
+                                            <div className="ui two column grid">
+                                                <div className="icon-align three wide column">
+                                                    <div />
+                                                    <i className="icon big usb"/>
+                                                    <div />
+                                                </div>
+                                                <div className="thirteen wide column">
+                                                    {lf("Download your code faster by pairing with web usb!")}
+                                                    <br/>
+                                                    <strong><a onClick={onPairClicked}>{lf("Pair now")}</a></strong>
+                                                </div>
+                                            </div>
+                                        </div>
+                                    </div>
+                                </div>
+                            </div>
+                            <div className="column">
+                                <div className="ui">
+                                    <div className="image">
+                                        <img alt={lf("Comic moving {1} file to {0}", boardDriveName, fileExtension)} className="ui medium rounded image" src={pxt.appTarget.appTheme.downloadDialogTheme?.dragFileImage} />
+                                    </div>
+                                </div>
+                            </div>
+                        </div>
+                    </div>
+                </div>
+            </div>
+            <div>
+                <sui.Checkbox
+                    inputLabel={lf("Don't show this again")}
+                    onChange={onCheckboxClicked}
+                />
+            </div>
+        </div>
+    </div>;
+}
+
+export function isDontShowDownloadDialogCookieSet() {
+    return pxt.BrowserUtils.getCookie(DONT_SHOW_COOKIE_ID) === "true";
```

Comment:
Can we use localStorage instead of cookie. This doesn't need to be cookie in my opionion.

Reply:
I mearging this PR now. But we should do this seperately.

Reply:
So what happens on a shared computer when a previous user has dismissed the dialogue?

It feels like this setting should be in memory or a user-settings is signed in.